module/grpcserver/server.go

```diff
@@ -0,0 +1,85 @@
+package grpcserver
+
+import (
+	"net"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"google.golang.org/grpc"
+
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+)
+
+// GrpcServer defines a grpc server that starts once and uses in different Engines.
+// It makes it easy to configure the node to use the same port for both APIs.
+type GrpcServer struct {
+	component.Component
+	log    zerolog.Logger
+	Server *grpc.Server
+
+	grpcListenAddr string // the GRPC server address as ip:port
+
+	addrLock    sync.RWMutex
+	grpcAddress net.Addr
+}
+
+// NewGrpcServer returns a new grpc server.
+func NewGrpcServer(log zerolog.Logger,
+	grpcListenAddr string,
+	grpcServer *grpc.Server,
+) (*GrpcServer, error) {
```

Comment:
nit: it doesn't look like an error is returned. let's remove it to avoid unnecessary error handler logic